Mathlib/Topology/UniformSpace/Dini.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Analysis.Normed.Order.Lattice
+import Mathlib.Topology.ContinuousMap.Ordered
+import Mathlib.Topology.UniformSpace.CompactConvergence
+
+/-! # Dini's Theorem
+
+This file proves Dini's theorem, which states that if `F n` is a monotone increasing sequence of
+continuous real-valued functions on a compact set `s` converging pointwise to a continuous function
+`f`, then `F n` converges uniformly to `f`.
+
+We generalize the codomain from `ℝ` to a normed latttice additive commutative group `G`.
+This theorem is true in a different generality as well: when `G` is a linearly ordered topological
+group with the order topology. This weakens the assumptions of the norm and commutativity, in
+exchange for strengthening to a linear order. This separate generality is not included in this file,
+but that generality was included in initial drafts of the original
+[PR #19068](https://github.com/leanprover-community/mathlib4/pull/19068) and can be recovered if
+necessary.
+
+-/
+
+open Filter Topology
+
+variable {ι α G : Type*} [Preorder ι] [IsDirected ι (· ≤ ·)] [Nonempty ι]
+variable [TopologicalSpace α] [NormedLatticeAddCommGroup G]
+
+section Unbundled
+
+open TopologicalAddGroup Metric
+
+variable {F : ι → α → G} {f : α → G}
+
+namespace Monotone
+
+/-- **Dini's theorem** If `F n` is a monotone increasing collection of continuous functions on a
+converging pointwise to a continuous function `f`, then `F n` converges locally uniformly to `f`. -/
+lemma tendstoLocallyUniformly_of_forall_tendsto
+    (hF_cont : ∀ i, Continuous (F i)) (hF_mono : Monotone F) (hf : Continuous f)
+    (h_tendsto : ∀ x, Tendsto (F · x) atTop (𝓝 (f x))) :
+    TendstoLocallyUniformly F f atTop := by
+  refine subsingleton_or_nontrivial G |>.elim (fun _ ↦ ?subsingleton) (fun _ ↦ ?_)
+  case subsingleton =>
+    rw [funext fun x => Subsingleton.elim (F x) f, TendstoLocallyUniformly]
+    -- missing `tendstoLocallyUniformly_const`
+    exact fun _ h x ↦ ⟨Set.univ, univ_mem, .of_forall fun _ _ _ ↦ mem_uniformity_of_eq h rfl⟩
+  have F_le_f (x : α) (n : ι) : F n x ≤ f x := by
+    refine _root_.ge_of_tendsto (h_tendsto x) ?_
```

Comment:
```suggestion
    refine ge_of_tendsto (h_tendsto x) ?_
```

Reply:
Can't do this because of `Monotone.ge_of_tendsto`, which should be protected.

Reply:
Well actually with the current imports there seems to be no issue!